File: leo2-node/src/main/java/org/deku/leo2/node/rest/services/internal/v1/TestService.java
package org.deku.leo2.node.rest.services.internal.v1;

import sx.rs.ApiKey;
import org.deku.leo2.rest.entities.internal.v1.TestEntry;

import javax.inject.Named;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import java.util.ArrayList;

/**
 * Created by masc on 23.07.14.
 */
@Named
@ApiKey(false)
@Path("internal/v1/test")
@Produces(MediaType.APPLICATION_JSON)
public class TestService implements org.deku.leo2.rest.services.internal.v1.TestService {
    @Override
    public TestEntry[] get() {
        ArrayList<TestEntry> entries = new ArrayList<>();
        for (int i = 0; i < 50; i++)
            entries.add(new TestEntry("test"));
        return entries.toArray(new TestEntry[0]);
    }

    @Override
    public TestEntry get(String name) {
        return null;
    }

    @Override
    public TestEntry[] find(String query) {
        return new TestEntry[0];
    }
}

File: leoz-node/src/main/java/org/deku/leoz/node/rest/service/internal/v1/TestService.kt
package org.deku.leoz.node.rest.service.internal.v1

import org.deku.leoz.ws.blz.BLZServicePortType
import org.slf4j.LoggerFactory
import sx.rs.auth.ApiKey
import javax.inject.Inject
import javax.inject.Named
import javax.ws.rs.Path
import javax.ws.rs.core.Response

/**
 * Created by masc on 09.10.15.
 */
@Named
@ApiKey(false)
@Path("internal/v1/test")
class TestService : org.deku.leoz.rest.service.internal.v1.TestService {
    private val log = LoggerFactory.getLogger(this.javaClass)

    @Inject
    private lateinit var blzService: BLZServicePortType

    @Inject
    private lateinit var glsShipmentProcessingService: org.deku.leoz.ws.gls.shipment.ShipmentProcessingPortType

    /**
     *
     */
    override fun testSoapCall(): Response {
        class Result(val result: String)

        val result = Result(
                blzService.getBank("50661639").bezeichnung)

        return Response
                .ok(result)
                .build()
    }
}